lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FocusEvent, KeyboardEvent, Ref, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import { always, dec, inc } from 'ramda';
+
+import './NxCombobox.scss';
+
+import forwardRef from '../../util/genericForwardRef';
+import { Props, propTypes } from './types';
+import NxTextInput from '../NxTextInput/NxTextInput';
+import NxDropdownMenu from '../NxDropdownMenu/NxDropdownMenu';
+import NxLoadError from '../NxLoadError/NxLoadError';
+import NxLoadingSpinner from '../NxLoadingSpinner/NxLoadingSpinner';
+import { useUniqueId } from '../../util/idUtil';
+import useToggle from '../../util/useToggle';
+import DataItem from '../../util/DataItem';
+export { Props } from './types';
+
+function NxComboboxRender<T extends string | number = string>(
+  props: Props<T>,
+  ref: Ref<HTMLDivElement>
+) {
+  const {
+        className: classNameProp,
+        loading,
+        loadError,
+        matches,
+        value,
+        onChange,
+        onSearch,
+        disabled,
+        emptyMessage,
+        autoComplete,
+        validatable,
+        isPristine,
+        validationErrors,
+        id,
+        'aria-required': ariaRequired,
+        'aria-describedby': ariaDescribedBy,
+        'aria-label': ariaLabel,
+        ...attrs
+      } = props,
+
+      isEmpty = !matches.length,
+      isAlert = loading || loadError || isEmpty,
+      dropdownRef = useRef<HTMLDivElement>(null),
+      inputRef = useRef<HTMLDivElement>(null),
+      alertRef = useRef<HTMLDivElement>(null),
+      showDropdown = !disabled && !isEmpty,
+      showAlert = !!(!disabled && isAlert && value),
+
+      [focusableBtnIndex, setFocusableBtnIndex] = useState<number | null>(null),
+      [inputIsFocused, toggleInputIsFocused] = useToggle(false),
+      [inlineStyle, setInlineStyle] = useState(false),
```

Comment:
I still feel like `inlineStyle` is a confusing variable name.  Calling it something like `showAutocomplete` would be accurate, right? That would feel clearer to me

Reply:
This variable means the autoComplete inline string has highlighted style, but seems like it is not that clear without explanation. I think it is correct to change it to the name that makes more senses. Updated here 5224a7116bb9da68f9d32ecc9476c370d0641910.